File: src/learning_ros_gyh21/src/learning_topic_vel_publisher.cpp
#include <ros/ros.h>
#include <geometry_msgs/Twist.h>

int main(int argc, char **argv)
{
    //ros节点初始化并创建句柄
    ros::init(argc, argv, "learn_topic_vel_puber");
    ros::NodeHandle nh;

    ros::Publisher turtle_vel_pub = nh.advertise<geometry_msgs::Twist>("/turtle1/cmd_vel", 10);

    ros::Rate loop_rate(10);

    int count = 0;
    while (ros::ok())
    {
        //初始化数据
        geometry_msgs::Twist vel_msg;
        vel_msg.linear.x = 0.5;
        vel_msg.angular.z = 0.2;
        //发送
        turtle_vel_pub.publish(vel_msg);

        ROS_INFO("Publish turtle vel command[%0.2f m/s %0.2f rad/s]", vel_msg.linear.x, vel_msg.angular.z);
        loop_rate.sleep();

        // 初始化，注册，封装数据，发送数据，循环
    }

    return 0;
}

File: src/learning_ros_gyh21/scripts/learning_topic_vel_publisher.py
#! /usr/bin/python
# -*- coding: utf-8 -*-
# 上边这句，python3不需要加
# 但是目前我们只能用python2,怎么用python3还得研究一下
import rospy
from geometry_msgs.msg import Twist

def vel_publisher():
    rospy.init_node('learning_topic_vel_puber',anonymous=True)
    turtle_vel_puber = rospy.Publisher('/turtle1/cmd_vel',Twist,queue_size=10)

    rate = rospy.Rate(10)
    while not rospy.is_shutdown():
        vel_msg = Twist()
        vel_msg.linear.x = 0.5
        vel_msg.angular.z = 0.2

        turtle_vel_puber.publish(vel_msg)
        rospy.loginfo('Publish turtle velocity command[%0.2f m/s, %0.2f rad/s]',vel_msg.linear.x,vel_msg.angular.z)

        rate.sleep()

if __name__ == '__main__':
    try:
        vel_publisher()
    except rospy.ROSInterruptException:
        pass